app/models/answer.rb

```diff
@@ -159,8 +164,23 @@ def option_node_id
 
   # This is a temporary method for assigning option based on an OptionNode ID.
   # Eventually Options will be removed and OptionNodes will be stored on Answers directly.
+  #
+  # Raises a loud error if the OptionNode is not in the OptionSet (or the mission) for security purposes.
   def option_node_id=(id)
-    self.option_id = id.present? ? OptionNode.id_to_option_id(id) : nil
+    self.option_id = if id.present?
+                       option_id = OptionNode.where(option_set: option_set).id_to_option_id(id)
+                       raise ArgumentError if option_id.nil?
+                       option_id
+                     end
```

Comment:
Note: As discussed, it would be more efficient to use `OptionNode.where(mission: mission)` for scoping (since option_set is delegated to questioning), but `response` isn't defined at this point so we can't use it to get the mission.

We could use `mission: option_set.mission` or `mission: questioning.mission` instead of `option_set: option_set` but that probably doesn't change much.